src/include/sof/probe/probe.h

```diff
@@ -12,6 +12,16 @@
 
 #include <ipc/probe.h>
 
+/*
+ * TODO:
+ *   - In IPC4, logging is enabled by attaching a probe
+ *     with ".probe_id.fields.module_id == 0".
+ *   - IPC3 does not have such a field, in IPC3 logging
+ *     prototype, a magic buffer identifier that is known
+ *     both by host and the FW, is used.
+ */
+#define PROBE_LOGGING_BUFFER_ID_MAGIC     999999
```

Comment:
If we decide to make a mergeable version for IPC3, perhaps the best approach would be extend the ".purpose" field of current probe point interface. I.e. you'd set up a probe point with a special purpose and this would be picked up FW to set up a probe point for logging. Any magic numbers could be hidden within the FW.

Reply:
What about using `BIT(31)` in buffer_id as a flag? If it is set then the rest of the buffer_id is ignored and it is for logging.
If you will:
```
union {
	uin32_t buffer_id;
	struct {
		uin32_t id : 31; /* buffer id if logging is 0 */
		uint32_t logging : 1;
	} fields;
};
```
or add new purpose, like `PROBE_PURPOSE_LOGGING` on top of EXTRACTION and INJECTION?

Reply:
New purpose added in V3 patch.